.github/workflows/build-test-images.yml

```diff
@@ -0,0 +1,41 @@
+name: build-test-images
+env:
+  DOCKERHUB_USER_NAME: ${{ secrets.DOCKERHUB_USER_NAME }}
+  DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
+  MAIN_BRANCH_NAME: master
+on:
+  push:
+    branches:
+      - "**"
+
+  workflow_dispatch:
+
+jobs:
+  build_package:
+    name: build-test-images
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USER_NAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+      # If current branch is not master,build and publish dev image
+      - name: Build & Push all dev images
+        if: ${{github.ref_name != env.MAIN_BRANCH_NAME }}
+        run: |
+          cd circleci/images
+          make push-all
```

Comment:
This job took 27 minutes on github actions. Which is a big reason of using github actions to begin with.

Does github actions support parallelism? There are many different images created, and they all have their own make target as well. Would it make sense to build them in parallel for development images?

Reply:
Yes it supports parallelism but as you said when we parallelize, each of the images will have different suffix. We can use sha as you suggest

Reply:
@thanodnl do you want me to parallelize the build images? AFAIU, reusable layers are not important for development images but important for images that CircleCI uses. If that's your point I can parallelize just the development images

Reply:
Yes please, as Talha also suggested, dev images benefit a lot from quick turnaround over layer caching, where release images benefit more from layer caching compared to quick turnaround.

Reply:
I parallelized the dev images as you suggested and used sha code as the suffix